augur/distance.py

```diff
@@ -51,18 +343,19 @@ def register_arguments(parser):
     parser.add_argument("--tree", help="Newick tree", required=True)
     parser.add_argument("--alignment", nargs="+", help="sequence(s) to be used, supplied as FASTA files", required=True)
     parser.add_argument('--gene-names', nargs="+", type=str, help="names of the sequences in the alignment, same order assumed", required=True)
+    parser.add_argument("--compare-to", nargs="+", choices=["root", "ancestor", "pairwise"], help="type of comparison between samples in the given tree", required=True)
+    parser.add_argument("--attribute-name", nargs="+", help="name to store distances associated with the given distance map", required=True)
+    parser.add_argument("--map", nargs="+", help="JSON providing the distance map between sites and, optionally, amino acids at those sites", required=True)
+    parser.add_argument("--date-annotations", help="JSON of branch lengths and date annotations from augur refine for samples in the given tree; required for comparisons by latest date")
+    parser.add_argument("--earliest-date", help="earliest date to consider samples for pairwise comparisons (e.g., 2019-01-01); if omitted, all samples prior to the latest date will be considered")
+    parser.add_argument("--latest-date", help="latest date to consider samples for last ancestor or pairwise comparisons (e.g., 2019-01-01); defaults to the current date")
     parser.add_argument("--output", help="JSON file with calculated distances stored by node name and attribute name", required=True)
-    parser.add_argument("--attribute-names", nargs="+", help="names to store distances associated with the corresponding masks", required=True)
-    parser.add_argument("--masks", help="tab-delimited mask definitions with mask name in first column and binary mask in second column")
-    parser.add_argument("--mask-names", nargs="*", help="name of each mask to use from the given masks file. Distances are calculated for each given mask and stored in each of the corresponding attribute names. If no mask is provided, all sites will be used.")
```

Comment:
This will be a pretty big breaking change. It looks like among our builds the command is only used by seasonal flu? Do we have any reason to think other, external folks are using `augur distance` and would benefit from backwards compat? If we decided it was worth it, It seems like it would be possible to automatically translate the old masks into the new maps and warn about deprecation.